timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -0,0 +1,366 @@
+                except ImportError:
+                    raise ImportError(
+                        "Huggingface Optimum library must be installed with ONNX for using the `onnx` strategy"
+                    )
+
+                assert kwargs.pop("device_map", "cpu") == "cpu", "ONNX mode only available on the CPU"
+                inner_model = ORTModelForSeq2SeqLM.from_pretrained(*args, **{**kwargs, "export": True})
+            elif optimization_strategy == "ovm":
+                try:
+                    from optimum.intel import OVModelForSeq2SeqLM
+                except ImportError:
+                    raise ImportError(
+                        "Huggingface Optimum library must be installed with OpenVINO for using the `ovm` strategy"
+                    )
+
+                inner_model = OVModelForSeq2SeqLM.from_pretrained(
+                    *args, **{**kwargs, "device_map": "cpu", "export": True}
+                )
+        else:
+            assert config.model_type == "causal"
+            inner_model = AutoModelForCausalLM.from_pretrained(*args, **kwargs)
+
+        return cls(
+            tokenizer=chronos_config.create_tokenizer(),
+            model=ChronosPretrainedModel(config=chronos_config, model=inner_model),
+        )
+
+
+class ChronosModel(AbstractTimeSeriesModel):
+    """Chronos pretrained time series forecasting models, based on the original
+    `ChronosModel <https://github.com/amazon-science/chronos-forecasting>`_ implementation.
+
+    Chronos is family of pretrained models, based on the T5 family, with number of parameters ranging between 8M and 710M.
+    The full collection of Chronos models is available on
+    `Hugging Face <https://huggingface.co/collections/amazon/chronos-models-65f1791d630a8d57cb718444>`_. For Chronos small,
+    base, and large variants a GPU is required to perform inference efficiently.
+
+    Chronos takes a minimalistic approach to pretraining time series models, by discretizing time series data directly into bins
+    which are treated as tokens, effectively performing regression by classification. This results in a simple and flexible framework
+    for using any language model in the context of time series forecasting. See [Ansari2024]_ for more information.
+
+    References
+    ----------
+    .. [Ansari2024] Ansari, Abdul Fatir, Stella, Lorenzo et al.
+        "Chronos: Learning the Language of Time Series."
+        http://arxiv.org/abs/2403.07815
+
+
+    Other Parameters
+    ----------------
+    model_path: str, default = "amazon/chronos-t5-small"
+        Model path used for the model, i.e., a HuggingFace transformers `name_or_path`. Can be a
+        compatible model name on HuggingFace Hub or a local path to a model directory.
+    batch_size : int, default = 16
+        Size of batches used during inference
+    num_samples : int, default = 20
+        Number of samples used during inference
+    skip_validation : bool, default = False
```

Comment:
I think this logic should be absorbed into a single boolean parameter that we pass to `TimeSeriesPredictor.fit` (currently `disable_length_check`). Otherwise, the user could write
```python
predictor.fit(data, hyperparameters={"Chronos": {"skip_validation": True}, "Naive": {}})
```
which would break our model selection logic.

The interpretation of the parameter passed to `TimeSeriesPredictor.fit` (I will call it `skip_validation`, but can be a different name) would be "I only want to train a single model and use it for prediction. I don't want to perform model selection, so I don't need to compute the validation score". If the user sets `predictor.fit(..., skip_validation=True)` but also provides >= 2 models via `hyperparameters` or provides a search space for any model, then an exception will be raised.

We could also split this functionality into a separate PR, and limit this PR to adding the Chronos model (that works like any other forecasting model).

What do you think?

Reply:
I agree to separating to a different PR.

Reply:
Okay, so just to make sure that we're on the same page, you will remove the `skip_validation` and `disable_length_check` arguments in this PR?